### PR TITLE
doc(join): add outer join

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
1. left outer join
2. right outer join
3. fix: queries can't produce results

## Changelog

- Documentation

## Related Issues

Fixes #issue

### CODE REVIEW

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md [hunk 1]
```diff
@@ -70,14 +72,14 @@ Table "purchase_records": This table lists the purchase records for all the clie
 The following command returns the purchase records of the VIP clients:
 
 ```sql    
-select * from vip_info inner join purchase_records on vip_info.Client_ID = purchase_records.Client_ID;
```
Comment (xudong963) [inline docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md:73]: https://github.com/datafuselabs/databend/issues/5236

**Comment by xudong963** [comment]
The pr should merge after https://github.com/datafuselabs/databend/pull/5972 is ready

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md [hunk 2]
```diff
@@ -181,4 +183,101 @@ Output:
...
+| 106         	| Soda      	| 4,000   	|
+
+The following command returns the purchase records of all VIP clients, the purchase records will be NULL if the VIP client has no purchases:
+
+```sql    
+select purchase_records.Client_ID, purchase_records.Item, purchase_records.QTY from vip_info left join purchase_records on vip_info.Client_ID = purchase_records.Client_ID;
+```
+Output:
+```sql
+|NULL|NULL|NULL
```
Comment (soyeric128) [inline docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md:231]: I guess the first NULL in the line should be 101

Comment (soyeric128) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md:231]: Oh i see why it's null. To return the purchase records of all VIP clients in this case, it seems we'd better select vip_info.Client_ID.

Comment (xudong963) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md]: > it seems we'd better select vip_info.Client_ID.

LGTM

**Review by soyeric128 — APPROVED** [review]

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md (hunk 2 above)
Comment (soyeric128) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md]: Thank you.

**Pull request merged**